toml/encoder.py

```diff
@@ -263,3 +280,50 @@ def __init__(self, _dict=dict, preserve=False):
         self.dump_funcs[np.float16] = _dump_float
         self.dump_funcs[np.float32] = _dump_float
         self.dump_funcs[np.float64] = _dump_float
+
+
+class TomlPreserveCommentEncoder(TomlEncoder):
+
+    def __init__(self, _dict=dict):
+        super(TomlPreserveCommentEncoder, self).__init__(_dict)
+
+    def write_comments(self, retval):
+        sections = retval.split('\n\n')
+        objects = []
+        for idx, content in enumerate(sections):
+            lst = content.split('\n')
+            for i, item in enumerate(lst):
+                split_pat = (re.findall(r'\"\s*\=\s*\"', item))
+                if split_pat:
+                    key = item.split(split_pat[0])
+                    if '-comment_inline' in key[0]:
+                        val = remove_quotes(key[1].strip())
+                        if val in key[0]:
+                            lst[i] = "blank"
+                            key = remove_quotes(key[0].
+                                                split('-comment_inline')[0])
+                            for indx, j in enumerate(lst):
+                                if len(j.split('=', 1)) == 2:
+                                    k = j.split('=')[0].strip()
+                                    if k == key:
+                                        val = unescape(val)
+                                        lst[indx] = j + ' ' + val
+                    elif '-comment' in key[0]:
```

Comment:
Will this have trouble with the following TOML?

```
a = 3
b-comment = "a is 3"
```

Reply:
No

Reply:
I have added these statements to test cases as well.

Reply:
This whole section sort of makes me nervous for a couple reasons:

1. It is dependent on a specific pattern of newlines to delineate "sections"
2. You are still saving the comments beside regular keys. At least, that is what it looks like to me? Do you have an example retval input for this `write_comments` function?